### PR TITLE
Array support for find + ResurrectHome: View Server Home Button on Server Guide

### PR DESCRIPTION
**Pull request by Nuckyz**
Adds a button to view the Server Home in the Server Guide channel row. Also adds support for array finds because it was needed to match the right module.

### CODE REVIEW

File: src/components/VencordSettings/PatchHelperTab.tsx [hunk 1]
```diff
@@ -180,15 +180,15 @@ function ReplacementInput({ replacement, setReplacement, replacementError }) {
 
     return (
         <>
-            <Forms.FormTitle>replacement</Forms.FormTitle>
+            <Forms.FormTitle className="">replacement</Forms.FormTitle>
```
Comment (Vendicated) [inline src/components/VencordSettings/PatchHelperTab.tsx:183]: is this empty className intentional?

File: src/plugins/resurrectHome/index.tsx [hunk 2]
```diff
@@ -92,14 +145,37 @@ export default definePlugin({
...
+            find: "69386d_5",
+            replacement: {
+                match: /location:"69386d_5".+?;/,
+                replace: "$&$self.useDisableViewServerHome();"
+            }
         }
     ],
 
+    ViewServerHomeButton: ErrorBoundary.wrap(({ text }: { text?: string; }) => {
+        if (text == null || text !== i18n.Messages.SERVER_GUIDE) return null;
```
Comment (Vendicated) [inline src/plugins/resurrectHome/index.tsx:168]:
```suggestion
        if (text !== i18n.Messages.SERVER_GUIDE) return null;
```

**Comment by Vendicated** [comment]
sooo this pr seems to patch the Guild thingie row component and check if `props.text` is GuildHome, then renders its own component

you could just directly patch the parent instead. the component takes a `badge` property that renders exactly in the position you want. you could just use that property and have a much cleaner patch without the hacky find for the super tiny generic module

image

File: src/components/VencordSettings/PatchHelperTab.tsx (hunk 1 above)
Comment (Nuckyz) [reply src/components/VencordSettings/PatchHelperTab.tsx:183]: yes, Discord codes adds a class I don't want when you dont have a className set

**Pull request merged**